subsys/net/lib/download_client/src/coap.c

```diff
@@ -24,6 +24,8 @@ extern char *strtok_r(char *str, const char *sep, char **state);
 int url_parse_file(const char *url, char *file, size_t len);
 int socket_send(const struct download_client *client, size_t len);
 
+static struct coap_block_context received_block_ctx;
```

Comment:
This will not work well in case of parallel downloads, I think we should rather store the information withing the download_client context.

Reply:
good point, thx